src/ServiceManagement/Compute/Commands.ServiceManagement.Preview/Generated/VirtualMachineVMImage/VirtualMachineVMImageUpdateMethod.cs

```diff
@@ -42,22 +42,22 @@ protected object CreateVirtualMachineVMImageUpdateDynamicParameters()
             {
                 ParameterSetName = "InvokeByDynamicParameters",
                 Position = 1,
-                Mandatory = true
+                Mandatory = false
             });
             pImageName.Attributes.Add(new AllowNullAttribute());
             dynamicParameters.Add("ImageName", pImageName);
 
             var pParameters = new RuntimeDefinedParameter();
-            pParameters.Name = "VirtualMachineVMImageUpdateParameters";
+            pParameters.Name = "VirtualMachineVMImageUpdateParameter";
             pParameters.ParameterType = typeof(VirtualMachineVMImageUpdateParameters);
             pParameters.Attributes.Add(new ParameterAttribute
             {
                 ParameterSetName = "InvokeByDynamicParameters",
                 Position = 2,
-                Mandatory = true
+                Mandatory = false
             });
             pParameters.Attributes.Add(new AllowNullAttribute());
-            dynamicParameters.Add("VirtualMachineVMImageUpdateParameters", pParameters);
+            dynamicParameters.Add("VirtualMachineVMImageUpdateParameter", pParameters);
```

Comment:
It seems this is impacting the Hyak side. The code generator needs to know about whether it's processing Hyak/AutoRest, and we would still need Hyak for ASM.